### PR TITLE
Merkle userid module: storing session data and domain

### PR DESCRIPTION
**Pull request by SKOCHERI**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/merkleIdSystem.js [hunk 1]
```diff
@@ -8,42 +8,80 @@
...
+  utils.logInfo('Merkle setting session ');
+  if (response && response.c && response.c.value && typeof response.c.value === 'string') {
+    setCookie(SESSION_COOKIE_NAME, response.c.value, storage.expires);
+  }
 }
 
-function constructUrl(configParams) {
-  const session = getSession(configParams);
+function constructUrl(configParams, storedId) {
+  const session = getSession(configParams, storedId);
```
Comment (msm0504) [inline modules/merkleIdSystem.js:44]: This 2nd parameter hasn't been added to the `getSession` function

**Review by msm0504 — CHANGES_REQUESTED** [review]

File: modules/merkleIdSystem.js [hunk 2]
```diff
@@ -8,42 +8,80 @@
...
   }
   return session;
 }
 
-function readCookie() {
-  return storage.cookiesAreEnabled() ? storage.getCookie(SESSION_COOKIE_NAME) : null;
+function setCookie(name, value, expires) {
+  let expTime = new Date();
+  expTime.setTime(expTime.getTime() + expires * 1000 * 60);
+  storage.setCookie(name, value, expTime.toGMTString());
```
[the remark this answers is not in the document]
Comment (SKOCHERI) [reply modules/merkleIdSystem.js]: Fixed

File: modules/merkleIdSystem.js (hunk 1 above)
Comment (SKOCHERI) [reply modules/merkleIdSystem.js]: Fixed

**Comment by jdwieland8282** [comment]
@msm0504 mind taking a look at this? @SKOCHERI made the requested changes and I'd like to merge this

**Review by jlukas79 — APPROVED** [review]

**Review by yava-ua — APPROVED** [review]

File: modules/merkleIdSystem.js (hunk 1 above)
Comment (jdwieland8282) [reply modules/merkleIdSystem.js]: @msm0504 mind approving your suggested changes have been implemented?

**Review by msm0504 — APPROVED** [review]

**Pull request merged**